crates/re_space_view_bar_chart/src/view_part_system.rs

```diff
@@ -24,17 +22,20 @@ impl NamedViewSystem for BarChartViewPartSystem {
 }
 
 impl ViewPartSystem for BarChartViewPartSystem {
-    fn archetype(&self) -> ArchetypeDefinition {
-        Tensor::all_components().try_into().unwrap()
+    fn required_components(&self) -> IntSet<ComponentName> {
+        Tensor::required_components()
+            .iter()
+            .map(ToOwned::to_owned)
+            .collect()
     }
 
-    fn queries_any_components_of(
+    fn heuristic_filter(
         &self,
         store: &re_arrow_store::DataStore,
         ent_path: &EntityPath,
-        components: &[ComponentName],
+        components: &IntSet<ComponentName>,
     ) -> bool {
-        if !components.contains(&re_types::components::TensorData::name()) {
+        if !components.contains(&re_types::archetypes::Tensor::indicator_component()) {
```

Comment:
Should barchart be its own Archetype?

Reply:
https://github.com/rerun-io/rerun/issues/3327